gyronimo/metrics/morphism.hh

```diff
@@ -21,22 +21,60 @@
 #define GYRONIMO_MORPHISM
 
 #include <gyronimo/core/IR3algebra.hh>
+#include <gyronimo/core/contraction.hh>
 
 namespace gyronimo {
 
 //! Abstract morphism from curvilinear (q) into cartesian (x) coordinates.
 class morphism {
- public:
-  virtual ~morphism() {};
-  virtual IR3 operator()(const IR3& q) const = 0;
-  virtual IR3 inverse(const IR3& x) const = 0;
-  virtual dIR3 del(const IR3& q) const = 0;
-  virtual dIR3 del_inverse(
-      const IR3& q) const {return gyronimo::inverse(del(q));};
-  virtual dIR3 tan_basis(const IR3& q) const {return del(q);};
-  virtual dIR3 dual_basis(const IR3& q) const {return del_inverse(q);};
-};
-
-}
+
+public:
+
+	//! Class Constructor
+	morphism() {};
+
+	//! Class Destructor
+	virtual ~morphism() {};
+
+	//! Maps the curvilinear coordinates `q` into cartesian coordinates `x`.
+	virtual IR3 operator()(const IR3 &q) const = 0;
+
+	//! Inverse transform from cartesian coordinates `x` into curvilinear coordinates `q`.
+	virtual IR3 inverse(const IR3 &x) const = 0;
+
+	//! Returns the morphism's derivatives, correspondent to the covariant basis vectors in point `q`.
+	virtual dIR3 del(const IR3 &q) const = 0;
+
+	//! Returns the morphism's inverse derivatives, correspondent to the contravariant basis vectors in point `q`.
+	virtual dIR3 del_inverse(const IR3 &q) const;
+
+	virtual dIR3 tan_basis(const IR3& q) const {return del(q);};
+	virtual dIR3 dual_basis(const IR3& q) const {return del_inverse(q);};
+
+	// //! Returns the morphism's second derivatives, calculated in point `q`.
+	// virtual dSM3 del2(const IR3& q) const = 0;
```

Comment:
If not needed, these lines should be removed.

Reply:
Taking care of it.